sim/midas/src/main/scala/midas/core/CPUManagedStreamEngine.scala

```diff
@@ -14,6 +14,54 @@ import freechips.rocketchip.util.DecoupledHelper
 import midas.targetutils.xdc
 import midas.widgets._
 
+class StreamAdapterIO(val w: Int) extends Bundle {
+  val in = Flipped(Decoupled(UInt(w.W)))
+  val out = Decoupled(UInt(w.W))
+
+  def flipConnect(other: StreamAdapterIO) {
+    in <> other.out
+    other.in <> out
+  }
+}
+
+class StreamWidthAdapter(narrowW: Int, wideW: Int) extends Module {
+  require(wideW >= narrowW)
+  require(wideW % narrowW == 0)
+  val io = IO(new Bundle {
+    val narrow = new StreamAdapterIO(narrowW)
+    val wide = new StreamAdapterIO(wideW)
+  })
+
+  if (wideW == narrowW) {
+    io.narrow.out <> io.wide.in
+    io.wide.out <> io.narrow.in
+  } else {
```

Comment:
If this isn't upstreamed to the CY version of this... can you do so?

Reply:
Will do later